source/common/event/user_space_file_event_impl.h

```diff
@@ -84,30 +83,27 @@ class UserSpaceFileEventImpl : public FileEvent, Logger::Loggable<Logger::Id::io
 
   EventListener& getEventListener() { return event_listener_; }
   void onEvents() { cb_(); }
-  friend class UserSpaceFileEventFactory;
+
+  // Helper method which is equivalent to active the pending events.
+  void scheduleNextEvent() {
+    if (!schedulable_->enabled()) {
+      schedulable_->scheduleCallbackNextIteration();
```

Comment:
scheduleCallbackNextIteration checks internally for enabled(), you can call it unconditionally.

Reply:
fixed